File: Crio/crio_programming_interview_problems-master/CheckBalanced/CheckBalanced.java
import java.util.*;
import crio.ds.Tree.TreeNode;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
public class CheckBalanced {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        for(int tests=0; tests<t;tests++)
        {
            TreeNode root  = TreeNode.readTreeReturnRoot(sc);
            boolean answer = new Solution().checkBalanced(root);
            if(answer) {
                System.out.println("Yes");
            } else {
                System.out.println("No");
            }
        }
    }
}

File: Crio/crio_programming_interview_problems-master/CheckBalanced/CheckBalanced.cpp
#include <bits/stdc++.h>
#include "../crio/cpp/ds/TreeNode/TreeNode.hpp"
#include "Solution.cpp"
using namespace std;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS


int main()
{
    int tests;
    cin >> tests;
    while (tests--)
    {
        TreeNode* root  = TreeNode().readTreeReturnRoot();
        bool answer = CheckBalanced().checkBalanced(root);
        if (answer)
            cout << "Yes\n";
        else
            cout << "No\n";
    }
    return 0;
}
